pilot/helpers/AgentConvo.py

```diff
@@ -54,7 +54,7 @@ def send_message(self, prompt_path=None, prompt_data=None, function_calls: Funct
 
         # TODO: move this if block (and the other below) to Developer agent - https://github.com/Pythagora-io/gpt-pilot/issues/91#issuecomment-1751964079
         # check if we already have the LLM response saved
-        if self.agent.__class__.__name__ == 'Developer':
+        if hasattr(self.agent, 'save_dev_steps') and self.agent.save_dev_steps:
```

Comment:
@LeonOstrez I believe you forgot this one in your recent refactor so added it here - LMK if this is incorrect.